File: src/python/problems/largest_odd_number_test.py
# https://leetcode.com/problems/largest-odd-number-in-string

from core.problem_base import *

class LargestOddNumber(ProblemBase):
    def Solution(self, num: str) -> str:
        for i in range(len(num) - 1, -1, -1):
            if int(num[i]) % 2 != 0:
                return num[:(i + 1)]

        return ""

if __name__ == '__main__':
    TestGen(LargestOddNumber) \
        .Add(lambda tc: tc.Param("52").Result("5")) \
        .Add(lambda tc: tc.Param("4206").Result("")) \
        .Add(lambda tc: tc.Param("35427").Result("35427")) \
        .Run()

File: src/challenges/rust/src/largest_odd_number_test.rs
// https://leetcode.com/problems/largest-odd-number-in-string

pub fn solution(num: String) -> String {
    let chars = num.as_bytes();
    for i in (0..num.len()).rev()  {
        if chars[i] % 2 != 0 {
            return String::from(&num[0..i + 1]);
        }
    }

    "".to_string()
}